arches/app/src/components/ControlledListManager/AddLabel.vue

```diff
@@ -0,0 +1,78 @@
+<script setup lang="ts">
+import arches from "arches";
+import { computed, ref } from "vue";
+import { useGettext } from "vue3-gettext";
+
+import EditLabel from "@/components/ControlledListManager/EditLabel.vue";
+
+import type {
+    ControlledListItem,
+    Label,
+    LanguageMap,
+    ValueType,
+} from "@/types/ControlledListManager.d";
+
+const props: {
+    item: ControlledListItem;
+    languageMap: LanguageMap;
+    type: ValueType;
+    onInsert: (label: Label) => Promise<Label>;
+} = defineProps(["item", "languageMap", "type", "onInsert"]);
+
+const visible = ref(false);
+
+const { $gettext } = useGettext();
+const slateBlue = "#2d3c4b"; // todo: import from theme somewhere
+
+const buttonLabel = computed(() => {
+    return props.type === "prefLabel"
+        ? $gettext("Add Preferred Label")
+        : $gettext("Add Alternate Label");
+});
+</script>
+
+<template>
+    <button
+        class="add-label"
+        @click="visible = true"
+    >
+        <i
+            class="fa fa-plus-circle"
+            aria-hidden="true"
+        />
+        <span class="add-label-text">
+            {{ buttonLabel }}
+        </span>
+    </button>
+    <EditLabel
+        v-model="visible"
+        :item
+        :header="buttonLabel"
+        :language-map
+        :label="{
+            id: null,
+            valuetype: props.type,
+            language: arches.activeLanguage,
```

Comment:
The way I've defined the types, neither `Label` nor `NewLabel` can lack  a language. I could make the EditLabel responsible for generating a new label, but I kind of preferred the explicitness of making the parent have a label in hand, and making a new one if need be.

But I don't like how verbose this is in the template, so I'll probably just extract this as a constant in the setup script. This suggests to me that the prop typing in EditLabel is not right, since it doesn't say it takes `NewLabel`. Will update.

Reply:
a690adffaa3559d63a1f99548522da22c53d9384